### PR TITLE
More responsive web frontend

### LINKED ISSUE #2250: [Feature]: serve web interface more mobile friendly
### Discussed in https://github.com/immich-app/immich/discussions/1650

<div type='discussions-op-text'>

<sup>Originally posted by **spupuz** October 26, 2022</sup>
### Feature detail

Please consider to make server web interface more mobile friendly

### Platform

Web</div>

### PR DESCRIPTION
**Pull request by faupau03**
Make web interface more responsive and easier to use on mobile devices.
Collapsable sidebar in user and admin interface,  search button and smaller server stats on mobile.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+PUBLIC_IMMICH_SERVER_URL=http://192.168.178.3:3001/api
```
Comment (alextran1502) [inline .env:1]: Can you remove this file?

File: web/src/hooks.server.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { env } from '$env/dynamic/public';
 import { ImmichApi } from './api/api';
 
 export const handle = (async ({ event, resolve }) => {
-	const basePath = env.PUBLIC_IMMICH_SERVER_URL || 'http://immich-server:3001';
+	const basePath = env.PUBLIC_IMMICH_SERVER_URL || 'http://192.168.178.3:3001';
```
Comment (alextran1502) [inline web/src/hooks.server.ts:7]: Can you revert this change?

File: web/vite.config.js [hunk 3]
```diff
@@ -13,8 +13,8 @@ const config = {
 		// connect to a remote backend during web-only development
 		proxy: {
 			'/api': {
-				target: process.env.PUBLIC_IMMICH_SERVER_URL,
-				secure: true,
+				target: 'http://192.168.178.3:3001',
```
Comment (alextran1502) [inline web/vite.config.js:16]: Can you revert changes in this file?

**Review by alextran1502 — CHANGES_REQUESTED** [review]
Thank you for the PR, I haven't tested the PR yet, just comment on some standout issues for now

File: .env (hunk 1 above)
Comment (faupau03) [reply .env]: Done

File: web/vite.config.js (hunk 3 above)
Comment (faupau03) [reply web/vite.config.js]: Done

**Review by alextran1502 — CHANGES_REQUESTED** [review]
Amazing work! thank you for such an important improvement on the UI.

A few things left to change

1. Can you help remove the `package-lock.json` file?
2. Can you change the upload button back to `LinkButton`?
3. Should we change the thumbnail size to be smaller on the small screen?

File: web/src/lib/components/shared-components/side-bar/side-bar.svelte [hunk 4]
```diff
@@ -69,9 +73,30 @@
...
+			isCollapsed = false;
+		} else {
+			isCollapsed = true;
+		}
+	};
+
+	onMount(() => {
+		handleResize();
+		window.addEventListener('resize', handleResize);
+	});
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/side-bar.svelte:88]: The event listener is added each time the component mounts, but it is never removed. Svelte provides a special element that can manage window event listeners for you. An example that also prevents using the global `innerWidth` variable:

```svelte
<script lang="ts">
	let innerWidth: number;

	onMount(() => {
		handleResize();
	});
</script>

<svelte:window on:resize={handleResize} bind:innerWidth />
```

Hardcoding the `md` breakpoint of 768px isn't that nice and the value might not make sense for some people without additional context. Also `@media (min-width: 768px)` means more or equal to `768` so you have an off-by-one error.

Alternatively this can probably be implemented with CSS only, using the Tailwind's `group` class. That will also prevent the layout shifting on desktop, because JS is executed later than CSS.

File: web/src/lib/components/shared-components/side-bar/side-bar.svelte [hunk 5]
```diff
@@ -69,9 +73,30 @@
...
 </script>
 
-<section id="sidebar" class="flex flex-col gap-1 pt-8 pr-6 bg-immich-bg dark:bg-immich-dark-bg">
+<section
+	id="sidebar"
+	on:mouseover={() => (innerWidth >= 430 ? (isCollapsed = false) : null)}
+	on:focus={() => null}
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/side-bar.svelte:94]: You probably added this empty event listener to stop a11y from complaining, but it would be preferred to correctly handle this situation by showing the extended sidebar on focus

File: web/src/routes/admin/+layout.svelte
Comment (michelheusschen) [inline web/src/routes/admin/+layout.svelte:1]: A lot of this is duplicated from `SideBar`, I recommend creating a new component instead

File: web/src/lib/components/layouts/user-page-layout.svelte [hunk 6]
```diff
@@ -23,7 +23,7 @@
 </header>
 
 <main
-	class="grid grid-cols-[250px_auto] relative pt-[4.25rem] h-screen bg-immich-bg dark:bg-immich-dark-bg immich-scrollbar"
+	class="grid md:grid-cols-[250px_auto] grid-cols-[70px_auto] relative pt-[4.25rem] h-screen bg-immich-bg dark:bg-immich-dark-bg immich-scrollbar"
```
Comment (michelheusschen) [inline web/src/lib/components/layouts/user-page-layout.svelte:26]: Using pixels for the grid layout means that zooming in and out will lead to issues, since Tailwind heavily uses relative units (rem). But if it's too difficult to change, no problem.

**Review by michelheusschen — COMMENTED** [review]
Thanks for your PR! I had this on my todo list but never got to it.

**Comment by faupau03** [comment]
> 



> Amazing work! thank you for such an important improvement on the UI.
> 
> A few things left to change
> 
>     1. Can you help remove the `package-lock.json` file?
> 
>     2. Can you change the upload button back to `LinkButton`?
> 
>     3. Should we change the thumbnail size to be smaller on the small screen?

1. Done, sorry it was added by wrongly using npm run dev
2. Done, i thought using icon button to set the title would make the upload button more accessible, especially if its icon only on small screens
3. I don't think thats necessary, if you mean immich logo without the text

File: web/src/lib/components/shared-components/side-bar/side-bar.svelte (hunk 5 above)
Comment (faupau03) [reply web/src/lib/components/shared-components/side-bar/side-bar.svelte:94]: Done

**Comment by jrasm91** [comment]
Any chance you could add some screenshots?

File: web/src/routes/admin/+layout.svelte
Comment (faupau03) [reply web/src/routes/admin/+layout.svelte:1]: Done in af168cc99f47665b893538876e0498d807a71b7b

File: web/src/lib/components/layouts/user-page-layout.svelte (hunk 6 above)
Comment (faupau03) [reply web/src/lib/components/layouts/user-page-layout.svelte:26]: I don't think thats something i can change easy, as i oriented myself on the 250px original sidebar size and just resized it to 70px. If you have an idea how to implement that in rem, i would be pleased to try my best on it. For now, i think its too difficult.

File: web/src/lib/components/shared-components/side-bar/side-bar.svelte (hunk 4 above)
Comment (faupau03) [reply web/src/lib/components/shared-components/side-bar/side-bar.svelte]: 1) Done in af168cc99f47665b893538876e0498d807a71b7b
2) I think for now hardcoding the breakpoint is the easiest way. I just oriented my self on how google has done it (they used max-width: 1007px).
3) I don't have used the Tailwind's `group` class up to now and don't know how to use it. I will try to read me in, mabye i can implement it that way, if not feel free to give it a go yourself.

P.S. Thanks for your feedback. As that is my first pull request, I am very interested and motivated in fixing the issues you mentioned.

**Comment by faupau03** [comment]
> Any chance you could add some screenshots?

Sure, here are some screenshots.  I choose two screen sizes, phone and desktop. Four screenshots of each size (light and dark, user and admin)
phone_light_admin
default_dark_user
default_light_user
default_dark_admin
default_light_admin
phone_light_user
phone_dark_user
phone_dark_admin

**Review by alextran1502 — APPROVED** [review]
